src/game_updater/GameUpdater.cpp

```diff
@@ -185,6 +186,14 @@ void GameUpdater::UpdatePlayer(std::shared_ptr<GameState> game_state) {
   } else {
     // player is in the air, apply gravity
     player->SetYVelocity(previous_player_velocity - PLAYER_GRAVITY);
+    // if player is close to falling, allow double jump
+    if (player->GetYVelocity() < PLAYER_JUMP_VELOCITY / 3.0 &&
+          player->GetDoubleJump() &&
+          ImGui::GetIO().KeysDown[GLFW_KEY_SPACE]) {
```

Comment:
If we get a system like joey suggested, where we can trigger events based off when keys are pressed rather than just checking if they're currently pressed, this if statement can be improved